engine/src/main/java/org/hibernate/search/v6poc/search/predicate/spi/SearchPredicateFactory.java

```diff
@@ -31,6 +31,6 @@
 
 	RangePredicateBuilder<C> range(String absoluteFieldPath);
 
-	NestedPredicateBuilder<C> nested(String absoluteFieldPath);
+	NestedPredicateBuilder<C> nested(String nestedPathContext, String absoluteFieldPath);
```

Comment:
Ok, this is not a good idea, really not. Think about what will happen if you create a `SearchPredicate`, then add it as a child of a new nested query. Then the "nested path context" of the child will be wrong.

We can't even work it around by creating specific instances of a specific `SearchPredicate` implementation whenever nested predicates are involved, since the `SearchPredicate` could also be a boolean predicate containing nested predicates...

First things first... could you add a test for this? Then we can try a few solutions.

I think that, if we want to solve this correctly, we will have to implement *all* `LuceneSearchPredicate`s differently: instead of containing a `Query`, they will contain a *factory* of `Query`, with a `create(String)` method accepting the nested path context.
The question is: how would we retrieve the nested path context from `LuceneSearchPredicate#contribute`... I guess we could add the `nestedPathContext` parameter to that method, instead of adding it to `SearchPredicateFactory#nested`.
What bugs me is that we would change a lot of code for a single specific case... Definitely not a great idea.

Another idea would be to systematically set the "nested path context" to the nearest `nested` parent in the metamodel, without even caring about the context in which the predicate appears. We can do that, fortunately, since we expect users to provide the absolute path of nested fields. Maybe that's even the reason why Elasticsearch expects the same.
I'm not sure if this would change anything to the results or to performance? I suspect it could be even better than what we have, since it would work even if the user skips some levels when "nesting" nested predicates (lvl1 > lvl3 for instance). But I may be wrong, and maybe in some cases it would not work at all...

I'm open to suggestions... Maybe it's better if we talk about it on the chat.